shared/actions/profile/proofs.js

```diff
@@ -79,147 +78,139 @@ const addServiceProof = (_, action: ProfileGen.AddProofPayload) =>
+        try {
+          const match = instructions.data.match(/<url>(http[s]+):\/\//)
+          const protocol = match && match[1]
+          actions.push(
+            Saga.put(ProfileGen.createUpdatePlatform({platform: protocol === 'https' ? 'https' : 'http'}))
           )
-        } else {
-          logger.info('Start Proof done: ', incoming.finished.params.sigID)
-          yield Saga.put(ProfileGen.createCheckProof())
+        } catch (_) {
+          actions.push(Saga.put(ProfileGen.createUpdatePlatform({platform: 'http'})))
         }
-        break
-      } else if (incoming['keybase.1.proveUi.promptOverwrite']) {
-        incoming['keybase.1.proveUi.promptOverwrite'].response.result(true)
-        // yield Saga.put(ProfileGen.createWaiting({waiting: true}))
-      } else if (incoming['keybase.1.proveUi.outputPrechecks']) {
-        incoming['keybase.1.proveUi.outputPrechecks'].response.result()
-        // yield Saga.put(ProfileGen.createWaiting({waiting: true}))
-      } else if (incoming['keybase.1.proveUi.preProofWarning']) {
-        incoming['keybase.1.proveUi.preProofWarning'].response.result(true)
-        // yield Saga.put(ProfileGen.createWaiting({waiting: true}))
-      } else if (incoming['keybase.1.proveUi.okToCheck']) {
-        incoming['keybase.1.proveUi.okToCheck'].response.result(true)
-        // yield Saga.put(ProfileGen.createWaiting({waiting: true}))
-      } else if (incoming['keybase.1.proveUi.displayRecheckWarning']) {
-        incoming['keybase.1.proveUi.displayRecheckWarning'].response.result()
-        // yield Saga.put(ProfileGen.createWaiting({waiting: true}))
       }
+
+      actions.push(Saga.put(ProfileGen.createUpdateProofText({proof})))
+      actions.push(
+        Saga.put(RouteTreeGen.createNavigateAppend({parentPath: [peopleTab], path: ['postProof']}))
+      )
+      return actions
+    }
+
+    const responseYes = (_, response) => response.result(true)
+
+    try {
+      const {sigID} = yield RPCTypes.proveStartProofRpcSaga({
+        customResponseIncomingCallMap: {
+          'keybase.1.proveUi.okToCheck': responseYes,
+          'keybase.1.proveUi.outputInstructions': outputInstructions,
+          'keybase.1.proveUi.preProofWarning': responseYes,
+          'keybase.1.proveUi.promptOverwrite': responseYes,
+          'keybase.1.proveUi.promptUsername': promptUsername,
+        },
+        incomingCallMap: {
+          'keybase.1.proveUi.displayRecheckWarning': () => {},
+          'keybase.1.proveUi.outputPrechecks': () => {},
+        },
+        params: {
+          auto: false,
+          force: true,
+          promptPosted: false,
+          service,
+          username: '',
+        },
+        waitingKey: Constants.waitingKey,
+      })
+      yield Saga.put(ProfileGen.createUpdateSigID({sigID}))
+      logger.info('Start Proof done: ', sigID)
+      yield Saga.put(ProfileGen.createCheckProof())
+    } catch (error) {
+      logger.warn('Error making proof')
```

Comment:
After erroring out of a service proof there's no way to continue
1. Start HN proof
2. Enter bogus username `abcdefgh`, 'Continue' -> error "Bad karma"
3. Switch to valid username `ayoubd`
4. 'Continue' doesn't do anything, need to back out and start over

Reply:
ok. this happens on master and i believe this is actually a core issue. discussing w/ patrick